articles/modules/ROOT/pages/how-to-force-match-ordering.adoc

```diff
@@ -0,0 +1,107 @@
+= How to Force Match Ordering
+:slug: how-to-force-match-ordering
+:author: Andrew Bowman
+:neo4j-versions: 3.5, 4.0, 4.1, 4.2, 4.3, 4.4, 5.x
+:tags: cypher
+:category: cypher
+
+Like SQL, Cypher is a declarative query language. This is most evident with Match patterns, which describe what you want to find in the graph.
+You do not dictate to it how to find these patterns, the way you might in an imperative programming language.
+
+Not only does the query planner decide how it will fulfill a single Match pattern, it has the ability to consider the entirety of patterns connected by common variables across multiple Match clauses in your query.
+This means that the planner does not have to fulfill Matches in the order that they occur in the query, and this can sometimes be surprising, especially when the ordering of operations the planner chooses to fulfill these Matches is suboptimal.
+
+Keep in mind also that Where clauses are not independent, and are bound to the preceding Match or With clause.
+This means that since multiple Match clauses are evaluated during planning, their Where clauses are likewise available for consideration.
+
+It is also important to understand that a With clause alone also cannot prevent the planner from considering Match patterns after it.
+
+This article discusses the reasons for this behavior, offers examples of when this behavior can become problematic, and provides techniques you can use to force the planner to fulfill separate Match clauses in the order they occur in your query.
+
+== In a query plan there is no Match, there is only Expand
+
+While the Cypher query describes "what" you want to find, the execution plan operators that make up a query plan are "how" Neo4j will fulfill the query.
+
+If you review https://neo4j.com/docs/cypher-manual/current/execution-plans/operator-summary/[query plan operators in the docs], you may note that they do not correspond one-to-one with Cypher clauses.
+There is no Match operator.
+Instead, to fulfill anchoring and expansion, there are several kinds of lookup operators (AllNodesScan, NodeByLabelScan, NodeIndexSeek, and others), and several kinds of expand operators.
+
+As such, this particular aspect of query planning involves the analysis of the Match patterns in a query (not Match pattern by Match pattern, but connected patterns across multiple Matches),
```

Comment:
try to keep the one sentence per line, to make it easier later to track edits.

Reply:
I'll make the changes, thanks!